### PR TITLE
[Enhancement](HttpServer) Add http interface authentication

### LINKED ISSUE #15658: [Enhancement] add http interface authentication and support https
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, doris supports access to fe / be only through http protocol, not https protocol. In addition, http does not perform interface authentication, which may cause security problems.

### Solution

Add http interface authentication and support https for doris, eliminate potential security risks and strengthen enterprise-level security support.
[Support https protocol and interface authentication](https://cwiki.apache.org/confluence/display/DORIS/DSIP-029%3A+Support+https+protocol+and+interface+authentication)

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)]

### PR DESCRIPTION
**Pull request by yongjinhou**
# Proposed changes

Issue Number: close #15658

## Problem summary

1. Organize http documents
2. **Add http interface authentication for FE**
3. Support https interface for FE
4. Provide authentication interface
5. Add http interface authentication for BE
6. Support https interface for BE

## Checklist(Required)

1. Does it affect the original behavior: 
    - [x] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.6 seconds
 stream load tsv:          477 seconds loaded 74807831229 Bytes, about 149 MB/s
 stream load json:         38 seconds loaded 2358488459 Bytes, about 59 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230209141421_clickbench_pr_93153.html

**Comment by carlvinhust2012** [comment]
maybe add some test case is more better.

**Comment by yangzhg** [comment]
After supporting authentication and https, can the internal access to these ports in doris still work normally?

**Comment by yongjinhou** [comment]
> maybe add some test case is more better.

Currently, there are no unit tests for http interface, I'll add one later.

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/BackendsAction.java [hunk 1]
```diff
@@ -70,9 +67,6 @@ public class BackendsAction extends RestBaseController {
     @RequestMapping(path = "/api/backends", method = {RequestMethod.GET})
     public Object getBackends(HttpServletRequest request, HttpServletResponse response) {
         executeCheckPassword(request, response);
-        if (Config.enable_all_http_auth) {
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/BackendsAction.java]: add some comments to explain why this api without auth

Comment (yongjinhou) [reply fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/BackendsAction.java]: I've explained that in the issue.

Comment (yongjinhou) [reply fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/BackendsAction.java]: Do I need to explain that in the code？

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run buildall

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**